soda/scientific/soda/scientific/distribution/comparison.py

```diff
@@ -240,15 +241,22 @@ def evaluate(self) -> Dict[str, float]:
 
 class SWDAlgorithm(DistributionAlgorithm):
     def evaluate(self) -> Dict[str, float]:
+        if any(isinstance(i, decimal.Decimal) for i in self.test_data):
+            self.test_data = self.test_data.astype("float")
+
         wd = wasserstein_distance(self.ref_data, self.test_data)
         swd = wd / np.std(np.concatenate([self.ref_data, self.test_data]))
         return dict(check_value=swd)
 
 
 class PSIAlgorithm(DistributionAlgorithm):
     def evaluate(self) -> Dict[str, float]:
+        if any(isinstance(i, decimal.Decimal) for i in self.test_data):
```

Comment:
what does this loop iterate over? is it rows?

Reply:
Yeah it's rows..

You would probably prefer something like

```if pd.core.dtypes.common.is_dtype_equal(self.test_data, Decimal)```

?

Also,  I'll move this type check higher up in the code to `DistributionChecker.run()`, as we run each test x times and currently convert `test_data` to float x times.

Reply:
Yeah I would like to see if we could find a method that doesn't force us to check each value. I would imagine that if any value in that column ends up being decimal then most likely the type of the column will not be `float` right? in that case then we apply. So maybe the function you refer to would work and I think it would be better provided it works as you intend it.

As to your second comment, I think I agree. It doesn't seem like the best idea to do this over and over again especially if it's costly.

Reply:
Yes good point!
I have tested `pd.core.dtypes.common.is_dtype_equal(self.test_data, Decimal)` and it returns `True` if there is at least one `decimal.Decimal`. If there are only values of type `float` then it returns `False`